libsolutil/Address.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsolutil/Address.h>
+#include <libsolutil/Exceptions.h>
+
+namespace solidity::util
+{
+
+/// @returns a h160 representation of an address string
+/// or an error message in case the address string is not a valid address
+Result<h160> validateAddress(std::string _addrString)
+{
+	std::string message;
```

Comment:
We are allowed to use namespace `std` in `cpp` files. You can add it to the top of the file and remove the qualifier.

```suggestion
Result<h160> validateAddress(string _addrString)
{
	string message;
```

Reply:
Changed.